identify: flaky TestIdentifyDeltaOnProtocolChange

https://github.com/libp2p/go-libp2p/runs/6609291832?check_suite_focus=true

```
 === RUN   TestIdentifyDeltaOnProtocolChange
      id_test.go:399: Missing id event
  --- FAIL: TestIdentifyDeltaOnProtocolChange (0.03s)

--- a/p2p/protocol/identify/id_test.go
+++ b/p2p/protocol/identify/id_test.go
@@ -395,7 +395,7 @@ func TestIdentifyDeltaOnProtocolChange(t *testing.T) {
 	case <-idComplete.Out():
 	case evt := <-idFailed.Out():
 		t.Fatalf("Failed to identify: %v", evt.(event.EvtPeerIdentificationFailed).Reason)
-	default:
+	case <-time.After(5 * time.Second):
 		t.Fatal("Missing id event")
 	}
 

